Sources/FaunaDB/QueryResult.swift

```diff
@@ -70,6 +95,41 @@ extension QueryResult {
 
 extension QueryResult {
 
+    /**
+        Apply a transformation if an error has occurred during the query execution.
+
+        If `mapErr` returns a value, the resulting `QueryResult` will be considered a success.
+        If you wish to handle an error but still return a failing `QueryResult`, you must rethrow
+        the original exception.
```

Comment:
I think technically saying 'the original exception' is a little too imprecise, as it technically doesn't indicate that you can throw any other exception to transform the error to another error. Although this seems obvious I think its slightly better to just say 'you must rethrow an exception'